### PR TITLE
Sync term signals

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

File: src/radical/pilot/agent/scheduler/base.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def initialize(self):
         self.register_subscriber(rpc.AGENT_UNSCHEDULE_PUBSUB, self.unschedule_cb)
 
         # start a process to host the actual scheduling algorithm
-        self._p = mp.Process(target=self._schedule_tasks)
+        self._p = mp.Process(target=self._schedule_tasks, args=(self._term,))
```
Comment (andre-merzky) [inline src/radical/pilot/agent/scheduler/base.py:258]: I don't think we need to pass `self._term` - `self` is passed implicitly (as the target is a member function) and that caries the `self._term` with it?

**Review by andre-merzky — COMMENTED** [review]
See comment.  Thanks!

File: src/radical/pilot/agent/scheduler/base.py (hunk 1 above)
Comment (mtitov) [reply src/radical/pilot/agent/scheduler/base.py:258]: yeah, this one is to pass term-signal explicitly, and thought to follow that `_subscribers` and `_publishers` are set within this `_schedule_tasks` method. Reassure environment setup for child process (?)

Comment (andre-merzky) [reply src/radical/pilot/agent/scheduler/base.py:258]: I think this is more confusing than helpful, as we then again overwrite `self._term` in the child and need to track that on debugging.

Comment (mtitov) [reply src/radical/pilot/agent/scheduler/base.py]: ok, reassignment removed

**Review by andre-merzky — APPROVED** [review]
Thanks!

**Pull request merged**